cni/pkg/ebpf/server/redirectionServer.go

```diff
@@ -0,0 +1,542 @@
+		log.Fatalf("Setting limit failed: %v", err)
+	}
+
+	r := &RedirectServer{
+		redirectArgsChan: make(chan *RedirectArgs),
+	}
+
+	r.initBpfObjects()
+
+	return r
+}
+
+func checkOrMountBPFFSDefault() error {
+	var err error
+
+	_, err = os.Stat(MapsRoot)
+	if err != nil {
+		if os.IsNotExist(err) {
+			if err := os.MkdirAll(MapsRoot, 0o755); err != nil {
+				return fmt.Errorf("unable to create bpf mount directory: %s", err)
+			}
+		}
+	}
+
+	fst := unix.Statfs_t{}
+	err = unix.Statfs(MapsRoot, &fst)
+	if err != nil {
+		return &os.PathError{Op: "statfs", Path: MapsRoot, Err: err}
+	} else if fst.Type == FilesystemTypeBPFFS {
+		return nil
+	}
+
+	err = unix.Mount(MapsRoot, MapsRoot, "bpf", 0, "")
+	if err != nil {
+		return fmt.Errorf("failed to mount %s: %s", MapsRoot, err)
+	}
+
+	return nil
+}
+
+func setLimit() error {
+	return unix.Setrlimit(unix.RLIMIT_MEMLOCK,
+		&unix.Rlimit{
+			Cur: unix.RLIM_INFINITY,
+			Max: unix.RLIM_INFINITY,
+		})
+}
+
+func (r *RedirectServer) Start(stop <-chan struct{}) error {
+	log.Infof("Starting redirection Server")
+	go func() {
+		for {
+			select {
+			case arg := <-r.redirectArgsChan:
+				if err := r.handleRequest(arg); err != nil {
+					log.Errorf("failed to handle request: %v", err)
+				}
+
+			case <-stop:
+				close(r.redirectArgsChan)
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+func (r *RedirectServer) handleRequest(redirectArgs *RedirectArgs) error {
+	var hw net.HardwareAddr
+	var multiErr error
+	var mapInfo mapInfo
+	ipAddr := redirectArgs.ipAddr
+	macAddr := redirectArgs.macAddr
+	ifindex := redirectArgs.ifindex
+	peerIndex := redirectArgs.peerIndex
+	ztunnel := redirectArgs.ztunnel
+	namespace := redirectArgs.namespace
+	remove := redirectArgs.remove
+
+	if !remove {
+		hw, multiErr = net.ParseMAC(macAddr)
+		if multiErr != nil {
+			return fmt.Errorf("invalid mac addr(%s): %v", macAddr, multiErr)
+		}
+		if len(hw) != 6 {
+			return fmt.Errorf("invalid mac addr(%s), only EUI-48/MAC-48 is supported", macAddr)
+		}
+		mapInfo.Ifindex = uint32(ifindex)
+		copy(mapInfo.MacAddr[:], hw)
+	}
+
+	if ztunnel {
+		if namespace == "" {
+			return fmt.Errorf("invalid namespace")
+		}
+		if remove {
+			if err := r.detachTCForZtunnel(uint32(ifindex), uint32(peerIndex), namespace); err != nil {
+				multiErr = concatError(multiErr, err)
+			}
+			// For array map, kernel doesn't support delete elem(refer to kernel/bpf/arraymap.c)
+			// it works just like an 'array'.
+			if err := r.obj.ZtunnelInfo.Update(uint32(0), mapInfo, ebpf.UpdateAny); err != nil {
+				multiErr = concatError(multiErr, err)
+			}
+		} else {
+			if err := r.attachTCForZtunnel(uint32(ifindex), uint32(peerIndex), namespace); err != nil {
+				multiErr = concatError(multiErr, err)
+			}
+			if err := r.obj.ZtunnelInfo.Update(uint32(0), mapInfo, ebpf.UpdateAny); err != nil {
+				multiErr = concatError(multiErr, err)
+			}
+		}
+	} else {
+		// ip slice is just in network endian
+		ip := net.ParseIP(ipAddr).To4()
```

Comment:
nit: would be preferred redirectArgs is typed and parsed outside, and used netip.Addr instead of the legacy `net`